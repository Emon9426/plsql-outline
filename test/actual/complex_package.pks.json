{
  "nodes": [
    {
      "label": "complex_package",
      "type": "package",
      "icon": "package",
      "children": [
        {
          "label": "MAX_VALUE",
          "type": "constant",
          "icon": "symbol-constant",
          "children": [],
          "range": {
            "startLine": 3,
            "endLine": 3
          }
        },
        {
          "label": "t_employee",
          "type": "type",
          "icon": "symbol-class",
          "children": [],
          "range": {
            "startLine": 6,
            "endLine": 6
          }
        },
        {
          "label": "t_employee_tab",
          "type": "type",
          "icon": "symbol-class",
          "children": [],
          "range": {
            "startLine": 12,
            "endLine": 12
          }
        },
        {
          "label": "invalid_value",
          "type": "exception",
          "icon": "error",
          "children": [],
          "range": {
            "startLine": 15,
            "endLine": 15
          }
        },
        {
          "label": "calculate_bonus",
          "type": "function",
          "icon": "symbol-function",
          "children": [],
          "range": {
            "startLine": 18,
            "endLine": 18
          }
        },
        {
          "label": "process_employees",
          "type": "procedure",
          "icon": "symbol-method",
          "children": [],
          "range": {
            "startLine": 21,
            "endLine": 21
          }
        },
        {
          "label": "format_name",
          "type": "function",
          "icon": "symbol-function",
          "children": [],
          "range": {
            "startLine": 24,
            "endLine": 24
          }
        }
      ],
      "range": {
        "startLine": 1,
        "endLine": 32
      }
    }
  ],
  "errors": []
}