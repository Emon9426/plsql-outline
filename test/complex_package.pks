/* 复杂包规范测试 */
CREATE OR REPLACE PACKAGE complex_package AS
  -- 常量声明
  MAX_VALUE CONSTANT NUMBER := 1000;
  
  -- 类型声明
  TYPE t_employee IS RECORD (
    id NUMBER,
    name VARCHAR2(100),
    salary NUMBER
  );
  
  TYPE t_employee_tab IS TABLE OF t_employee INDEX BY PLS_INTEGER;
  
  -- 异常声明
  invalid_value EXCEPTION;
  
  -- 函数声明
  FUNCTION calculate_bonus(p_salary NUMBER, p_factor NUMBER) RETURN NUMBER;
  
  -- 过程声明
  PROCEDURE process_employees(p_emp_tab IN OUT t_employee_tab);
  
  -- 嵌套函数声明
  FUNCTION format_name(p_name VARCHAR2) RETURN VARCHAR2;
  
  -- 注释中的关键字测试：BEGIN END FUNCTION PROCEDURE DECLARE
  /* 
   * 这是一个多行注释
   * 包含关键字：BEGIN 和 END
   * 以及 FUNCTION, PROCEDURE
   */
END complex_package;
/
