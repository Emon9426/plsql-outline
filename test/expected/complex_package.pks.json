{
  "nodes": [
    {
      "label": "complex_package",
      "type": "package",
      "icon": "package",
      "children": [],
      "range": {
        "startLine": 1,
        "endLine": 1
      }
    }
  ],
  "errors": []
}